File: javascript/resistor-color-duo/resistor-color-duo.js
export const value = (colorInputs) => {

  const totalString = colorInputs.map(i => `${COLORS.indexOf(i)}`).join("")

  return Number(totalString)
};

const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
    @colorArray = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
    
    def self.value(colors)
        colors.map { |colour| @colorArray.index(colour) }.join.to_i
    end
end